tests/test_agents.py

```diff
@@ -1059,7 +1059,34 @@ async def test_clinical_tool_usage(agent_test_settings) -> None:
     assert any(
         "ClinicalTrials.gov" in c.text.doc.citation for c in response.session.contexts
     ), "No clinical trials were put into contexts"
-
+    
+@pytest.mark.asyncio
+async def test_search_pagination(agent_test_settings: Settings) -> None:
+    """Test that pagination works correctly in SearchIndex.query()."""
+    
+    index = await get_directory_index(settings=agent_test_settings)
+    
+    page_size = 1
+    
+    results1 = await index.query(
+        query="test", 
+        top_n=page_size,
+        offset=0
+    )
+    results2 = await index.query(
+        query="test", 
+        top_n=page_size,
+        offset=page_size
+    )
+    all_results = await index.query(
+        query="test",
+        top_n=2*page_size,
+        offset=0
+    )
```

Comment:
```suggestion
    page1_results = await index.query(
        query="test", 
        top_n=page_size
    )
    page2_results = await index.query(
        query="test", 
        top_n=page_size,
        offset=page_size
    )
    page1and2_results = await index.query(
        query="test",
        top_n=2*page_size
    )
```

Let's be a bit more explicit in the variable names here

I find "results1" is not really as intuitive as "page1_results?